### PR TITLE
:sparkle: Improved OpenAI API Key Insert to Env

### PR DESCRIPTION
**Pull request by Qoyyuum**
### Background
Relates to #2148 but most commonly to fix a lot of users trying to use the AutoGPT and failed to set up their `.env` file or to set up the environment variable properly.

### Changes
Improved `check_openai_api_key()` function

### Documentation
Documented in the function itself. It's a small singular atomic function that handles checking and having the user input into the environment variable and `Config` of AutoGPT.

### Test Plan
Manually run the function. Unable to write a unit test script that can handle user input.

### PR Quality Checklist
- [X] My pull request is atomic and focuses on a single change.
- [X] I have thoroughly tested my changes with multiple different prompts.
- [X] I have considered potential risks and mitigations for my changes.
- [X] I have documented my changes clearly and comprehensively.
- [X] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Review by Pwuts — COMMENTED** [review]
Does it also write back to the .env file?

File: autogpt/config/config.py [hunk 1]
```diff
@@ -269,4 +270,14 @@ def check_openai_api_key() -> None:
             + "Please set your OpenAI API key in .env or as an environment variable."
         )
         print("You can get your key from https://platform.openai.com/account/api-keys")
-        exit(1)
+        openai_api_key = input(
+            "If you do have the key, please enter your OpenAI API key now:\n"
```
Comment (Pwuts) [inline autogpt/config/config.py:315]: Maybe clarify that it will not be saved (and people will still need to set it in `.env` for repeated use)

Comment (Qoyyuum) [reply autogpt/config/config.py:315]: Good idea. I'll edit my PR to include that

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

File: autogpt/config/config.py (hunk 1 above)
Comment (k-boikov) [reply autogpt/config/config.py:315]: @Qoyyuum Any news here?

Comment (Qoyyuum) [reply autogpt/config/config.py:315]: Sorry took awhile. I've just committed with a new message that should be clearer for new users.

**Comment by lc0rp** [comment]
Are we ok with putting this in place temporarily? It's probably worth waiting for #4604 to be implemented so that we don't change things for users too often.

I'll flag it for discussion.

**Comment by Pwuts** [comment]
> Are we ok with putting this in place temporarily? It's probably worth waiting for https://github.com/Significant-Gravitas/Auto-GPT/issues/4604 to be implemented [...]

@lc0rp, that depends on the timeline for #4604 :)

**Review by lc0rp — APPROVED** [review]

**Pull request merged**